src/bci_build/package/__init__.py

```diff
@@ -1343,10 +1351,21 @@ class DevelopmentContainer(BaseContainerImage):
 
     def __post_init__(self) -> None:
         super().__post_init__()
+
+        if self.version and not self.tag_version:
+            self.tag_version = self.version
+
+    def prepare_generation(self) -> None:
+        """Hook to do delayed expensive work prior template rendering"""
+        super().prepare_generation()
+
         if not self.version:
-            raise ValueError("A language stack container requires a version")
+            raise ValueError("A development container requires a version")
+
         if not self.tag_version:
-            self.tag_version = self.version
+            raise ValueError("A development container requires a tag_version")
```

Comment:
Imho this can stay in `__post_init__`. This check is super cheap and should fail as early as possible